### DISCUSSION TITLE
Running an experiment in several languages

### DISCUSSION THREAD

**Discussion by yseulthb** [jspsych/jsPsych#1868] [Q&A] [answered]
Hello! 
First of all, I want to express my gratitude towards jsPsych; I've just started using it, this is so convenient!
I have a question regarding the possibility of running an experiment in several languages (all using the Latin alphabet, if that matters).
I want to give the participants the option to choose their preferred language between 3 options at the beginning of the experiment. I can think of a conditional loop, but that would mean having three blocks of instruction and every other variable that displays text. 
Is there any alternative that someone might have thought of? Or is it possible to include a conditional loop within the instruction block, for example? 
Thanks for your input!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @yseulthb, great question. You can do this using [dynamic parameters](https://www.jspsych.org/overview/dynamic-parameters/) so that you only need to create the instruction trial once (and any other trial that includes text), and make any text-related parameters dynamic, i.e. functions that return the appropriate value, based on the language that was selected. 

There are lots of ways to do this, but here's one example. In this example I've created a global variable called `lang`. The value is `null` when the experiment starts, and then it's set in the `on_finish` function of the trial where the participant selects their preferred language. This way, the `lang` variable can be used throughout the whole experiment to determine what text-related parameter values should be used. Note that the `lang` variable must be used inside of a dynamic parameter or [event-related callback function](https://www.jspsych.org/overview/callbacks/).

_(Apologies to all Spanish speakers for errors!)_

```js
  // create a global variable to hold the language selection
  var lang = null;

  var preferred_language = {
    type: 'survey-multi-choice',
    questions: [{
      prompt: '<p>What is your preferred language? /</p>¿Cual es tu idioma preferido?', 
      options: ['English/Inglés','Spanish/Español'], 
      name: 'lang', 
      required: true
    }],
    on_finish: function(data) {
      // set the value of the lang variable based on the participant's choice
      if (data.response.lang == "English/Inglés") {
        lang = "eng";
      } else {
        lang = "spa";
      }
    }
  };

  var instructions = {
    type: 'instructions',
    pages: function() {
      // use a dynamic parameter to use different instruction text strings, depending on the current value of the lang variable
      if (lang == "eng") {
        return ['Welcome!', 'In this experiment...']
      } else if (lang == "spa") {
        return ['¡Bienvenido!', 'En este experimento...']
      }
    },
    show_clickable_nav: true,
    button_label_next: function() {
      // use a dynamic parameter to use different button labels, depending on the current value of the lang variable
      if (lang == "eng") {
        return "Next";
      } else if (lang == "spa") {
        return "Próximo";
      }
    },
    button_label_previous: function() {
      if (lang == "eng") {
        return "Back";
      } else if (lang == "spa") {
        return "Atrás";
      }
    }
  };

  jsPsych.init({
    timeline: [preferred_language, instructions], 
    on_finish: function(){jsPsych.data.displayData();}
  });
```

**Reply by yseulthb** [reply]
thank you, that's exactly what I was looking for! it works perfectly!
